scripts/update-venue-owners.ts

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node -r esm -r ts-node/register
+
+import {
+  checkFileExists,
+  FieldValue,
+  findUserByEmail,
+  initFirebaseAdminApp,
+  makeScriptUsage,
+  parseCredentialFile,
+} from "./lib/helpers";
+
+// ---------------------------------------------------------
+// Configuration (this is the bit you should edit)
+// ---------------------------------------------------------
+
+const VENUES_TO_CHANGE_OWNERSHIP: string[] = [];
+
+const EMAIL_ADDRESSES: string[] = [];
+
+// ---------------------------------------------------------
+// HERE THERE BE DRAGONS (edit below here at your own risk)
+// ---------------------------------------------------------
+
+enum Mode {
+  Add = "ADD",
+  Remove = "REMOVE",
+}
+
+const VALID_MODES: string[] = [Mode.Add, Mode.Remove];
+
+const usage = makeScriptUsage({
+  description: `Bulk add users to the owners list based on the supplied email address(es).\n\n(Note: the accounts must already exist.)`,
+  usageParams: `CREDENTIAL_PATH [${Mode.Add}/${Mode.Remove}] EMAIL1 EMAIL2 EMAIlN...`,
+  exampleParams: `fooAccountKey.json ADD foouser@example.com baruser@example.com`,
+});
+
+const [credentialPath, mode, ...extraEmailAddresses] = process.argv.slice(2);
+if (!credentialPath || !VALID_MODES.includes(mode)) {
+  usage();
+}
+
+const allEmailAddresses = [...EMAIL_ADDRESSES, ...extraEmailAddresses];
+
+if (allEmailAddresses.length === 0) {
+  console.error("Error: No email addresses provided");
+  process.exit(1);
+}
+
+if (VENUES_TO_CHANGE_OWNERSHIP.length === 0) {
+  console.error("Error: No venues provided");
+  process.exit(1);
+}
+
+if (!checkFileExists(credentialPath)) {
+  console.error("Credential file path does not exists:", credentialPath);
+  process.exit(1);
+}
+
+const { project_id: projectId } = parseCredentialFile(credentialPath);
+
+if (!projectId) {
+  console.error("Credential file has no project_id:", credentialPath);
+  process.exit(1);
+}
+
+const app = initFirebaseAdminApp(projectId, { credentialPath });
+
+(async () => {
+  for (const venue of VENUES_TO_CHANGE_OWNERSHIP) {
```

Comment:
It might be good to use the batch write pattern here to do all of the updates in one atomic write. There should be an example of this in a script I worked on recently, I think it might have been the venue access one? Might still be in an unnerved PR

Reply:
Instead of looping through things with for like this, we tend to use .map, .filter, etc